src/audio/crossover/crossover.c

```diff
@@ -642,215 +665,179 @@ static int crossover_copy(struct comp_dev *dev)
 	buffer_release(source_c);
 
-	return ret;
+	list_for_item(sink_list, &dev->bsink_list) {
+		sinkb = container_of(sink_list, struct comp_buffer, source_list);
+		sink_c = buffer_acquire(sinkb);
+		ipc4_update_buffer_format(sink_c, &mod->priv.cfg.base_cfg.audio_fmt);
+		buffer_release(sink_c);
+	}
 }
+#endif
 
 /**
  * \brief Prepares Crossover Filter component for processing.
  * \param[in,out] dev Crossover Filter base component device.
  * \return Error code.
  */
-static int crossover_prepare(struct comp_dev *dev)
+static int crossover_prepare(struct processing_module *mod,
+			     struct sof_source __sparse_cache **sources, int num_of_sources,
+			     struct sof_sink __sparse_cache **sinks, int num_of_sinks)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_data *cd =  module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 	struct comp_buffer *source, *sink;
 	struct comp_buffer __sparse_cache *source_c, *sink_c;
 	struct list_item *sink_list;
-	int32_t sink_period_bytes;
-	int ret;
+	int channels;
+	int ret = 0;
 
 	comp_info(dev, "crossover_prepare()");
 
-	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
-	if (ret < 0)
-		return ret;
-
-	if (ret == COMP_STATUS_STATE_ALREADY_SET)
-		return PPL_STATUS_PATH_STOP;
+#if CONFIG_IPC_MAJOR_4
+	crossover_params(mod);
+#endif
 
 	/* Crossover has a variable number of sinks */
-	source = list_first_item(&dev->bsource_list,
-				 struct comp_buffer, sink_list);
+	mod->max_sinks = SOF_CROSSOVER_MAX_STREAMS;
+	source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
 	source_c = buffer_acquire(source);
 
 	/* Get source data format */
 	cd->source_format = audio_stream_get_frm_fmt(&source_c->stream);
+	channels = audio_stream_get_channels(&source_c->stream);
+	audio_stream_init_alignment_constants(1, 1, &source_c->stream);
+	buffer_release(source_c);
 
 	/* Validate frame format and buffer size of sinks */
 	list_for_item(sink_list, &dev->bsink_list) {
 		sink = container_of(sink_list, struct comp_buffer, source_list);
 		sink_c = buffer_acquire(sink);
-
 		if (cd->source_format != audio_stream_get_frm_fmt(&sink_c->stream)) {
-			comp_err(dev, "crossover_prepare(): Source fmt %d and sink fmt %d are different for sink %d.",
-				 cd->source_format, audio_stream_get_frm_fmt(&sink_c->stream),
-				 sink_c->pipeline_id);
+			comp_err(dev, "crossover_prepare(): Source fmt %d and sink fmt %d are different.",
+				 cd->source_format, audio_stream_get_frm_fmt(&sink_c->stream));
 			ret = -EINVAL;
-		} else {
-			sink_period_bytes = audio_stream_period_bytes(&sink_c->stream,
-								      dev->frames);
-			if (audio_stream_get_size(&sink_c->stream) < sink_period_bytes) {
-				comp_err(dev,
-					 "crossover_prepare(), sink %d buffer size %d is insufficient",
-					 sink_c->pipeline_id, audio_stream_get_size(&sink_c->stream));
-				ret = -ENOMEM;
-			}
 		}
 
+		audio_stream_init_alignment_constants(1, 1, &sink_c->stream);
```

Comment:
this needs to be done even if the formats didn't match and we're going to error out?

Reply:
Yep, ordered this way for more compact appear with my style but done now.